### PR TITLE
build: update multiple modules for spotbug 4.3.0 and re-enable plugin

### PR DESCRIPTION
**Pull request by mjsax**
Follow up to #7808

Remaining packages:
cli
examples
metastore
rest-client
serde
streams
test-util
tools
udf
version-metrics-clients

### CODE REVIEW

File: ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/KsqlMaterialization.java [hunk 1]
```diff
@@ -213,7 +213,7 @@ public MaterializedWindowedTable windowed() {
   }
 
   /*
-  Today, we are unconditionally adding the extra fields to windowed rows.
+   Today, we are unconditionally adding the extra fields to windowed rows.
```
Comment (mjsax) [inline ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/KsqlMaterialization.java:216]: Side cleanup

File: ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/Row.java [hunk 2]
```diff
@@ -80,6 +81,7 @@ public GenericKey key() {
   }
 
   @Override
+  @SuppressFBWarnings(value = "EI_EXPOSE_REP", justification = "should be mutable")
```
Comment (mjsax) [inline ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/Row.java:84]: If not mutable, `KsqlMaterializationTest` breaks. (Wondering is this is really intentional?)

File: ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/WindowedRow.java [hunk 3]
```diff
@@ -88,6 +89,7 @@ public GenericKey key() {
   }
 
   @Override
+  @SuppressFBWarnings(value = "EI_EXPOSE_REP", justification = "should be mutable")
```
Comment (mjsax) [inline ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/WindowedRow.java:92]: If not mutable, `KsqlMaterializationTest` breaks. (Wondering is this is really intentional?)

**Review by stevenpyzhang — APPROVED** [review]
gave a quick skim, lgtm if build passes

**Pull request merged**